src/core/reference/include/openvino/reference/detection_output.hpp

```diff
@@ -366,12 +366,25 @@ class referenceDetectionOutput {
     void mxNetNms(const LabelBBox& decodeBboxesImage,
                   const std::map<int, std::vector<dataType>>& confScores,
                   std::map<int, std::vector<int>>& indices) {
+        const auto is_background_label = [this](int c) {
+            return attrs.background_label_id > -1 && c == attrs.background_label_id;
+        };
         std::vector<std::pair<dataType, std::pair<int, int>>> scoreIndexPairs;
         for (size_t p = 0; p < numPriors; p++) {
             dataType conf = -1;
             int id = 0;
-            for (int c = 1; c < numClasses; c++) {
-                if (attrs.background_label_id > -1 && c == attrs.background_label_id)
+            int c = 1;
+            if constexpr (std::is_unsigned_v<dataType>) {
+                for (; c < numClasses; ++c) {
+                    if (is_background_label(c))
+                        continue;
+                    conf = confScores.at(c)[p];
+                    id = c++;
+                    break;
+                }
+            }
```

Comment:
Can you explain which issue it fix?
It looks like for unsigned type the different start values of `id, c` are calculated but then old code is executed, how it fix coverity issue?

Reply:
Yes, it addresses CIDs 2356337, 2358440. The case was that for unsigned the `conf` was initialized with type max value, what lead to dead-code. For signed types it picks up first biggest bigger than -1 from the set (except the background label).

Reply:
This change was removed from the PR as unsigned types should not be evaluated for this Op.